### PR TITLE
feat: implement listagg macro

### PR DESCRIPTION
**Pull request by igorvoltaic**
## Summary
Implement listagg macro, which is not supported at the moment

### CODE REVIEW

File: dbt/include/clickhouse/macros/utils/utils.sql [hunk 1]
```diff
@@ -63,11 +63,15 @@
...
-{% macro clickhouse__listagg(measure, delimiter_text, order_by_clause, limit_num) %}
-  {{ exceptions.raise_compiler_error(
-    'ClickHouse does not support the listagg function.  See the groupArray function instead')
-    }}
-{% endmacro %}
+{% macro clickhouse__listagg(measure, delimiter_text, order_by_clause, limit_num) -%}
+    {% set arr = "array_agg({})".format(measure) %}
+    {% set arr = "arraySort({}, {})".format(arr, order_by_clause) if order_by_clause else arr %}
+    {% if limit_num -%}
+      arrayStringConcat('{{delimiter_text}}', arraySlice({{ arr }}, 1, {{ limit_num }}))
```
Comment (BentsiLeviav) [inline dbt/include/clickhouse/macros/utils/utils.sql:70]: based on the [docs](https://clickhouse.com/docs/en/sql-reference/functions/splitting-merging-functions#arraystringconcat) it seems that you need to pass the column name/array first, and only then the delimiter.

Comment (igorvoltaic) [reply dbt/include/clickhouse/macros/utils/utils.sql:70]: ah, thanks a lot!

Comment (igorvoltaic) [reply dbt/include/clickhouse/macros/utils/utils.sql]: based on [docs](https://docs.getdbt.com/reference/dbt-jinja-functions/cross-database-macros#listagg) singlequote also not needed. fixed that too

File: dbt/include/clickhouse/macros/utils/utils.sql [hunk 2]
```diff
@@ -63,11 +63,15 @@
...
 
 
-{% macro clickhouse__listagg(measure, delimiter_text, order_by_clause, limit_num) %}
-  {{ exceptions.raise_compiler_error(
-    'ClickHouse does not support the listagg function.  See the groupArray function instead')
-    }}
-{% endmacro %}
+{% macro clickhouse__listagg(measure, delimiter_text, order_by_clause, limit_num) -%}
+    {% set arr = "array_agg({})".format(measure) %}
+    {% set arr = "arraySort({}, {})".format(arr, order_by_clause) if order_by_clause else arr %}
```
Comment (BentsiLeviav) [inline dbt/include/clickhouse/macros/utils/utils.sql:68]: The order by part doesn't work. You can validate the entire macro with the test's result.

Comment (igorvoltaic) [reply dbt/include/clickhouse/macros/utils/utils.sql:68]: Thanks again. Now, I'm not sure how I checked this in the first place

I've considered two options here:
```
arraySlice(arrayMap(x -> (x.1), arraySort(x -> (x.2), arrayZip(array_agg(a), array_agg(d)))), 1, 10)

<Debug> executeQuery: Read 50000000 rows, 238.42 MiB in 8.861773 sec., 5642211.778613603 rows/sec., 26.90 MiB/sec.
<Debug> MemoryTracker: Peak memory usage (for query): 1.23 GiB.
```

and 

```
select  arraySlice(mapValues(mapSort(mapFromArrays(array_agg(d), array_agg(a)))), 1,10) from t;

<Debug> executeQuery: Read 50000000 rows, 238.42 MiB in 15.139353 sec., 3302651.0445988015 rows/sec., 15.75 MiB/sec.
<Debug> MemoryTracker: Peak memory usage (for query): 1.23 GiB.
```
Opted for the first one with the results

**Comment by BentsiLeviav** [comment]
I created a different task (https://github.com/ClickHouse/dbt-clickhouse/issues/301) for adjusting the tests to assert the results.

**Pull request merged**